scripts/setup_database.php

```diff
@@ -86,8 +86,11 @@
 if (strcasecmp($session_type, 'DB')==0) {
     printf("Creating database table hm_user_session ...\n");
 
-    if ($db_driver == 'mysql' || $db_driver == 'sqlite') {
+    if ($db_driver == 'mysql') {
         $stmt = "{$create_table} hm_user_session (hm_id varchar(255), data longblob, date timestamp, primary key (hm_id));";
+    } elseif($db_driver == 'sqlite') {
+        //0 means unlocked, 1 means locked 
+        $stmt = "{$create_table} hm_user_session (hm_id varchar(255), data longblob, lock INTEGER DEFAULT 0, date timestamp, primary key (hm_id));";
```

Comment:
How would existing users update their databases? I think we should introduce some kind of a migration mechanism or handle this in the code.

Reply:
The way you did it is a bit fragile and would require careful maintenance of the existing db schema as a php array which might be a problem for programmers. I'd suggest we use a migration-kind of mechanism where we have initial schema and then migration files for each change we do - much like Laravel, Rails and other similar frameworks work. The migrations can be simple sql files in a special directory where the main schema resides. Then a simple script can be added to migrate - checking which ones are already executed and executing the rest. We should have a table to track the executed migrations, so some kind of ID or migration file name should be recorded in that table.

Reply:
oh i see. I'l work on that.